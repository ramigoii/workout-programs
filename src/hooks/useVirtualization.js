import { useState, useEffect } from 'react';

export const useVirtualization = (initialColumnCount = 3) => {
  const [columnCount, setColumnCount] = useState(initialColumnCount);
  const [dimensions, setDimensions] = useState({
    width: window.innerWidth,
    height: window.innerHeight,
  });

  useEffect(() => {
    const handleResize = () => {
      setDimensions({
        width: window.innerWidth,
        height: window.innerHeight,
      });

      if (window.innerWidth < 640) {
        setColumnCount(1);
      } else if (window.innerWidth < 1024) {
        setColumnCount(2);
      } else {
        setColumnCount(initialColumnCount);
      }
    };

    handleResize(); // Set initial values
    window.addEventListener('resize', handleResize);
    return () => window.removeEventListener('resize', handleResize);
  }, [initialColumnCount]);

  return { columnCount, dimensions };
};

export default useVirtualization;
