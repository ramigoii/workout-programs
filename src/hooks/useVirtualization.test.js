import { renderHook, act } from '@testing-library/react';
import { useVirtualization } from './useVirtualization';

// Mock window resize
const resizeWindow = (width) => {
  window.innerWidth = width;
  window.dispatchEvent(new Event('resize'));
};

describe('useVirtualization', () => {
  test('returns correct column count based on screen width', () => {
    const { result } = renderHook(() => useVirtualization(3));
    
    // Initial state
    expect(result.current.columnCount).toBe(3);

    // Small screen
    act(() => resizeWindow(600));
    expect(result.current.columnCount).toBe(1);

    // Medium screen
    act(() => resizeWindow(800));
    expect(result.current.columnCount).toBe(2);

    // Large screen
    act(() => resizeWindow(1200));
    expect(result.current.columnCount).toBe(3);
  });

  test('snapshot of hook result', () => {
    const { result } = renderHook(() => useVirtualization(3));
    expect(result.current).toMatchSnapshot();
  });
});
