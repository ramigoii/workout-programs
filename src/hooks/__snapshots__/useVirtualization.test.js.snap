// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useVirtualization snapshot of hook result 1`] = `
Object {
  "columnCount": 3,
  "dimensions": Object {
    "height": 768,
    "width": 1200,
  },
}
`;
